### DISCUSSION TITLE
Why does Homebrew change the path for the Xcode command line tools?

### DISCUSSION THREAD

**Discussion by hovsater** [Homebrew/discussions#167] [Everyday usage] [answered]
I just had a fresh install of macOS. I installed Xcode followed by installing Homebrew. During the installation, Homebrew went ahead and installed another set of CLT located at `/Library/Developer/CommandLineTools`. I confirmed this by running `xcode-select --print-path` and it reported `/Library/Developer/CommandLineTools` as expected.

Now, I also do iOS & RN development and during a `pod install`, `glog` failed to build with the following output:

```
checking whether the C compiler works... no
xcrun: error: SDK "iphoneos" cannot be located
xcrun: error: SDK "iphoneos" cannot be located
xcrun: error: SDK "iphoneos" cannot be located
xcrun: error: unable to lookup item 'Path' in SDK 'iphoneos'
```

I opened Xcode and went into Preferences -> Locations and noted no CLT version was selected in the dropdown. I went a head and set it to `Xcode 12.2 (12B45b)`. Reran `pod install` and everything compiled just fine.

Now I noticed that this had the side effect of changing the path from `/Library/Developer/CommandLineTools` to `/Applications/Xcode.app/Contents/Developer`.

Is this a problem? Does Homebrew expect it to point to `/Library/Developer/CommandLineTools`?

Would love some clarity on the matter. 🙂

**Comment by SMillerDev** [comment, marked as answer]
Homebrew needs the CLT and sometimes it also needs Xcode. Once you install Homebrew it pulls in the CLT and that install changes the path Xcode recognizes for the CLT. Unfortunately Apple made this difficult because the CLT Xcode provides and the CLT that macOS can install independently aren't the same.

This means that on your next brew usage homebrew might complain that it can't find stuff in the CLT.

**Reply by hovsater** [reply]
@SMillerDev I see. Does that mean I should switch back to `/Library/Developer/CommandLineTools`?

So Xcode wants its own bundled CLT and Homebrew wants the standalone ones? Does this mean I would have to manually switch between the two whenever I interact with Homebrew or an iOS project that require dependencies to be compiled? I guess what's a bit confusing still is why we need both?

By the looks of it, Xcode does not play nicely when the path is set to `/Library/Developer/CommandLineTools`, so I'll need to set it to `/Applications/Xcode.app/Contents/Developer` to make it work. But at the same time, my understanding was that this may break Homebrew?

Is this really by design?

**Reply by SMillerDev** [reply]
> But at the same time, my understanding was that this may break Homebrew?

It may, but largely brew knows how to find it's tools pretty well. Basically you can keep it set to Xcode and if you encounter weird brew bugs I'd look at this first.

> Is this really by design?

Well, brew and Apple are two seperate entities. Brew just has to work with the hand Apple deals us. I'm sure you can relate to that as an iOS developer :smile:

**Reply by hovsater** [reply]
@SMillerDev haha, 💯! Thanks for the clarification. I'll just keep this in mind moving forward. 🙂

Thanks a lot for taking the time to answer this. Appreciate it!
